auditor/src/routes/record_handlers.rs

```diff
@@ -12,10 +11,28 @@ pub struct RecordQuery {
 
 #[tracing::instrument(name = "Getting records", skip(query, pool))]
 pub async fn query_records(
-    query: Option<QsQuery<Filters>>,
+    query: HttpRequest,
     pool: web::Data<PgPool>,
 ) -> Result<HttpResponse, GetFilterError> {
-    let records = advanced_record_filtering(query.as_ref(), &pool)
+    let query_string = query.query_string();
+
+    let filters: Filters = serde_qs::from_str(query_string).unwrap();
```

Comment:
I think we should also return an error here (probably also 400) instead of calling `unwrap()` (which would kill the server process).